rasa/nlu/selectors/response_selector.py

```diff
@@ -338,21 +349,31 @@ def _full_response(self, label: Dict[Text, Any]) -> Optional[Dict[Text, Any]]:
             the text but also buttons, images, ...
         """
         for key, responses in self.responses.items():
-            for response in responses:
-                if hash(response.get(TEXT, "")) == label.get("id"):
-                    return response
-        return None
+            if self.train_on_text:
+                for response in responses:
+                    if hash(response.get(TEXT, "")) == label.get("id"):
+                        return key, responses
```

Comment:
why does this need to return a tuple, can't the caller also use `self.responses[key]`?

Reply:
Refactored the whole function to be cleaner.